fetch.bs

```diff
@@ -2940,67 +2940,99 @@ run these steps:</p>
 
 <h3 id=cross-origin-resource-policy-header>`<code>Cross-Origin-Resource-Policy</code>` header</h3>
 
-<p>The
+<p>Servers can use the
 `<dfn export http-header id=http-cross-origin-resource-policy><code>Cross-Origin-Resource-Policy</code></dfn>`
-response <a for=/>header</a> can be used to require checking a <a for=/>request</a>'s
-<a for=request>current URL</a>'s <a for=url>origin</a> against a <a for=/>request</a>'s
-<a for=request>origin</a> when <a for=/>request</a>'s <a for=request>mode</a> is
-"<code>no-cors</code>".
+response <a for=/>header</a> to control the contexts into which a given resource may be loaded. The
+restrictions asserted by this header apply to "<code>no-cors</code>" subresource requests and to
+<a>navigation requests</a> that target a frame or a window that's accessible from the request's
+initiator.
 
 <p>Its <a for=header>value</a> <a>ABNF</a>:
 
 <pre><code class=lang-abnf>
-Cross-Origin-Resource-Policy     = %s"same-origin" / %s"same-site" ; case-sensitive
+Cross-Origin-Resource-Policy     = %s"same-origin" / %s"same-site" / %s"cross-site" ; case-sensitive
 </code></pre>
 
 <p>To perform a <dfn>cross-origin resource policy check</dfn>, given a <var>request</var> and
 <var>response</var>, run these steps:</p>
 
 <ol>
- <li><p>If <var>request</var>'s <a for=request>mode</a> is not "<code>no-cors</code>", then return
- <b>allowed</b>.
-
  <li>
-  <p>If <var>request</var>'s <a for=request>origin</a> is <a>same origin</a> with
-  <var>request</var>'s <a for=request>current URL</a>'s <a for=url>origin</a>, then return
-  <b>allowed</b>.
+  <p>Return <b>allowed</b> if either of the following statements is true:
+  <ul class=brief>
+   <li><p><var>request</var>'s <a for=request>mode</a> is "<code>same-origin</code>",
+   "<code>cors</code>", or "<code>websocket</code>".
 
-  <p class="note no-backref">While redirects that carry  a
-  `<a http-header><code>Cross-Origin-Resource-Policy</code></a>` header are checked, redirects
-  without such a header resulting in <var>response</var> do not contribute to this algorithm. I.e.,
-  <var>request</var>'s <a for=request>tainted origin flag</a> is not checked.
+   <li><p><var>request</var> is a <a>navigation request</a> whose <a for=request>reserved client</a>
+   is an <a for=/>environment</a> whose <a for=environment>target browsing context</a> is neither a
+   <a>nested browsing context</a> nor an <a>auxiliary browsing context</a>.
```

Comment:
This should explain why the auxiliary navigating the non-auxiliary is not a problem.

Reply:
Well, hrm. I guess that actually is a problem without process isolation. `attacker.site` could open `attacker.site` in a new window, the latter could navigate the former to `victim.site`, and sadness would ensue. Perhaps this should instead ask whether we're navigating within a unit of related browsing contexts? Or whatever the new hotness is? "User agent cluster"?

Reply:
Without process isolation on the agent cluster level, right.

I think the navigation checks only make sense when the flag is set on a document. And that flag only makes sense if COOP is also set. (COOP guaranteeing process isolation on the browsing context group level.)

Reply:
New hotness background:

* https://html.spec.whatwg.org/#integration-with-the-javascript-agent-cluster-formalism
* https://html.spec.whatwg.org/#groupings-of-browsing-contexts
* https://github.com/whatwg/html/issues/4361